### PR TITLE
homed: map container UID range for homed home directories

### PR DESCRIPTION
**Pull request by poettering**
Inspired by, based on, and replacing #22239 by @brauner.

It's a very similar patch, but uses append_identity_range() to generate the range (so that in the unlikely event we are not actually using UID_NOBODY on the backing fs but one from the container range things will work reasonably OK). Also, and more importantly this maps the container range we already document in https://systemd.io/UIDS-GIDS/

### CODE REVIEW

**Review by brauner — APPROVED** [review]

File: docs/UIDS-GIDS.md [hunk 1]
```diff
@@ -233,6 +233,27 @@ safely use the NSS user database as allocation check, too. Note that if you
...
+files/directories in home directories managed by `systemd-homed` cannot be
+owned by UIDs/GIDs outside of these four ranges (attempts to `chown()` files to
+UIDs outside of these ranges will fail). Thus, if container trees are to be
+placed within a home directory managed by `systemd-homed` they should take
+these ranges into consideration and either place the trees at base UID 0 (and
+then map them to a higher UID range for use in user namespacing via another
+level of UID mapped mounts, at *runtime*) or at a base UID from the container
+UID range. That said, placing container trees (and in fact any
+files/directories not owned by the home directory's user) in home directories
+is generally a questionnable idea (regardless if `systemd-homed` is used or
```
Comment (bluca) [inline docs/UIDS-GIDS.md]: `questionable`
`regardless of whether`

File: docs/UIDS-GIDS.md [hunk 2]
```diff
@@ -233,6 +233,27 @@ safely use the NSS user database as allocation check, too. Note that if you
...
+not), given this typically breaks quota assumptions, makes it impossible for
+users to properly manage all files in their own home directory due to
+permission problems, introduces security issues around SETUID and severly
```
Comment (bluca) [inline docs/UIDS-GIDS.md]: `severely`

**Review by bluca — APPROVED** [review]

File: src/home/homework-mount.c [hunk 3]
```diff
@@ -209,6 +209,13 @@ static int make_userns(uid_t stored_uid, uid_t exposed_uid) {
         if (r < 0)
                 return log_oom();
 
+        /* Also map the container range. People can user that to place containers owner by high UIDs in their
```
Comment (yuwata) [inline src/home/homework-mount.c]: `can user` -> `can use`?

**Comment by poettering** [comment]
Addressed all raised points. No other changes. Upgrading label.

**Pull request merged**